### PR TITLE
Prevent duplicate text in workspace folder picker when path has no parent (fix #183418)

### LINKED ISSUE #183418: Workspace folder picker entry descriptions are suboptimal for some filesystems
Type: <b>Bug</b>

1. Configure setting `"terminal.integrated.splitCwd": "workspaceRoot"` in order to provide a convenient way of invoking the workspace folder picker by splitting a terminal when in a multiroot workspace.
2. Install the MemFS extension (jrieken.vscode-memfs) from Marketplace.
3. Close any open folder or workspace, then run the `MemFS: Setup Workspace` command.
4. Run `Preferences: Open Workspace Settings (JSON)`
5. Edit this to simulate a multi-root workspace in which the paths of the uris point to the filesystem root of two different authorities:
```json
{
	"folders": [
		{
			"name": "MemFS - Sample 1",
			"uri": "memfs://1/"
		},
		{
			"name": "MemFS - Sample 2",
			"uri": "memfs://2/"
		}
	],
	"settings": {}
}
```
6. On the Terminal tab on the Panel, use the Split Terminal button.

The Workspace folder picker looks like this:

image

Displaying the folder name twice on each line is unhelpful.

 When the path of the uri is `/` I propose showing the non-relative label instead. I will submit a PR.

VS Code version: Code - Insiders 1.79.0-insider (f6be5461f8bc69013a605f5baea834651c6589fb, 2023-05-20T01:18:53.811Z)
OS version: Windows_NT x64 10.0.22621
Modes:

### PR DESCRIPTION
**Pull request by gjsjohnmurray**
This PR fixes #183418, which contains repro steps suitable for testing.

### CODE REVIEW

**Review by bpasero — CHANGES_REQUESTED** [review]
This change seems strange to me. I think the intent here is to not show `description` when the result matches that of `label`. I would prefer a solution that computes `label` and `description` and then compares both and ignoring `description` when they are the same. Anything else is strange path magic that we should stay away from imho.

**Comment by gjsjohnmurray** [comment]
Pushed a change which gives this for the testcase:

image

File: src/vs/workbench/browser/actions/workspaceCommands.ts [hunk 1]
```diff
@@ -117,9 +117,14 @@ CommandsRegistry.registerCommand(PICK_WORKSPACE_FOLDER_COMMAND_ID, async functio
 	}
 
 	const folderPicks: IQuickPickItem[] = folders.map(folder => {
+		const label = folder.name;
```
Comment (bpasero) [inline src/vs/workbench/browser/actions/workspaceCommands.ts:120]: My suggestion:

```ts
const folderPicks: IQuickPickItem[] = folders.map(folder => {
		const label = folder.name;
		const description = labelService.getUriLabel(dirname(folder.uri), { relative: true });

		return {
			label,
			description: description !== label ? description : undefined, // https://github.com/microsoft/vscode/issues/183418
			folder,
			iconClasses: getIconClasses(modelService, languageService, folder.uri, FileKind.ROOT_FOLDER)
		};
	});
```

I think we can have `description` undefined to hide it?

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/actions/workspaceCommands.ts (hunk 1 above)
Comment (gjsjohnmurray) [reply src/vs/workbench/browser/actions/workspaceCommands.ts:120]: Good idea. Change pushed.

**Review by bpasero — APPROVED** [review]

**Review by sandy081 — APPROVED** [review]

**Pull request merged**